lib/dry/system/stub.rb

```diff
@@ -0,0 +1,27 @@
+module Dry
+  module System
+    class Container
+      # Incuded only in the Test environment
+      # Sending the message enable_stubs! allow you to stub components after
+      # finalize your container in your tests.
+      #
+      # @api private
+      module Stub
+        def finalize!(&block)
+          yield(self) if block
+
+          importer.finalize!
+          booter.finalize!
+          manual_registrar.finalize!
+          auto_registrar.finalize!
```

Comment:
Your point about this method duplicating al the calls of the main `.finalize!` method is a good one.

Feels like it'd be _ok_ if we added a `freeze:` boolean kwarg to the main `.finalize!` and just pass `super(freeze: true, &block)` like you mention.  I think we'd want to make that kwarg part of our "private" API though, since we'd want people to use `enable_stubs!` instead of manually passing `freeze: true`.